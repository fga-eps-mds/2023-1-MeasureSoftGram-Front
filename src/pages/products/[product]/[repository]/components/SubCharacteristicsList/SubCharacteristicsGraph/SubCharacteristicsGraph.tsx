import React, { useState } from 'react';

import { TableCell, TableRow, Collapse } from '@mui/material';
import ArrowCircleRightIcon from '@mui/icons-material/ArrowCircleRight';
import ArrowCircleDownIcon from '@mui/icons-material/ArrowCircleDown';

import GraphicStackedLine from '@components/GraphicStackedLine';
import { Historical, SqcHistory } from '@customTypes/repository';

interface OptionCheckedProps {
  [key: string]: boolean;
}

interface Props {
  SQC: SqcHistory;
  checkedOptions: OptionCheckedProps;
  subCharacteristics: Historical[];
}

function SubCharacteristicsGraph({ SQC, checkedOptions, subCharacteristics }: Props) {
  const [open, setOpen] = useState(false);
  const { value, created_at: createdAt } = SQC;

  return (
    <div data-testid="sub-characteristics-graph">
      <TableRow data-testid="open-row" hover onClick={() => setOpen(!open)} sx={{ '& > *': { borderBottom: 'unset' } }}>
        <TableCell data-testid="open">
          {open ? <ArrowCircleDownIcon aria-label="expand row" /> : <ArrowCircleRightIcon aria-label="expand row" />}
        </TableCell>
        <TableCell align="right">{value.toFixed(2)}</TableCell>
      </TableRow>
      <TableRow>
        <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={3} align="right">
          <Collapse in={open}>
            <GraphicStackedLine
              historical={subCharacteristics}
              checkedOptions={checkedOptions}
              title="Sub-CaracterÃ­ticas"
              selected={(item: Date) => createdAt === item}
            />
          </Collapse>
        </TableCell>
      </TableRow>
    </div>
  );
}

export default SubCharacteristicsGraph;
