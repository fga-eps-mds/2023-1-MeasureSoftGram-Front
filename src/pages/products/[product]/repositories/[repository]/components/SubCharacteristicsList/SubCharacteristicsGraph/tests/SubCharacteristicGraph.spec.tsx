import '@testing-library/jest-dom';

import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import SubCharacteristicsGraph from '../SubCharacteristicsGraph';

const SQC = {"id":302,"value":0.6641230995964292,"created_at":new Date("2022-08-28T18:04:48")}
const checkedOptions = {"reliability":true,"maintainability":true,"testing_status":true,"modifiability":true}
const repositoryHistoricalSubCharacteristics = [{"id":1,"key":"modifiability","name":"Modifiability","description":null,"history":[{"id":577,"subcharacteristic_id":1,"value":0.6454758461579367,"created_at":new Date("2022-08-01T18:31:00")}]}]

jest.mock('echarts-for-react', () => ({
  __esModule: true,
  ...jest.requireActual('echarts-for-react'),
  default: () => <div />
}));

describe('<SubCharacteristicsGraph />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = render(
        <SubCharacteristicsGraph
          SQC={SQC}
          checkedOptions={checkedOptions}
          subCharacteristics={repositoryHistoricalSubCharacteristics}
        />
      )

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Comportamento', () => {
    it('Deve corresponder ao Snapshot', () => {
      const { getByTestId } = render(
        <SubCharacteristicsGraph
          SQC={SQC}
          checkedOptions={checkedOptions}
          subCharacteristics={repositoryHistoricalSubCharacteristics}
        />
      )

      fireEvent.click(getByTestId('open-row'))

      expect(getByTestId('open')).toBeInTheDocument();
    });
  });
});
