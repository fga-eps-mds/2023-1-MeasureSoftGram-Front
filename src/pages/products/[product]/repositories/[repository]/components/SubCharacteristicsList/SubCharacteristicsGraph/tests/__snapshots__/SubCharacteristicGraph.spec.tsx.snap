// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubCharacteristicsGraph /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="sub-characteristics-graph"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-hover css-2blr9s-MuiTableRow-root"
          data-testid="open-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
            data-testid="open"
          >
            <svg
              aria-hidden="true"
              aria-label="expand row"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowCircleRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zm-10 1H8v-2h4V8l4 4-4 4v-3z"
              />
            </svg>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeMedium css-r7oxtk-MuiTableCell-root"
          >
            0.66
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeMedium css-r7oxtk-MuiTableCell-root"
            colspan="3"
            style="padding-bottom: 0px; padding-top: 0px;"
          >
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    class="styles__GraphicContainer-sc-1nisug5-0 glYmeV"
                  >
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="sub-characteristics-graph"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-2blr9s-MuiTableRow-root"
        data-testid="open-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="open"
        >
          <svg
            aria-hidden="true"
            aria-label="expand row"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zm-10 1H8v-2h4V8l4 4-4 4v-3z"
            />
          </svg>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeMedium css-r7oxtk-MuiTableCell-root"
        >
          0.66
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeMedium css-r7oxtk-MuiTableCell-root"
          colspan="3"
          style="padding-bottom: 0px; padding-top: 0px;"
        >
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  class="styles__GraphicContainer-sc-1nisug5-0 glYmeV"
                >
                  <div />
                </div>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
